File: pset1/greedy.c
#include<cs50.h>
#include<math.h>
#include<stdio.h>
int main()
{
    float f;
    int sum,quarter,dime,nickel,rem,total=0;
    do{
        printf("O hai! How much change is owed?\n");
        f=get_float();
        sum=round(f*100);
        quarter=sum/25;
        rem=sum%25;
        dime=rem/10;
        rem=rem%10;
        nickel=rem/5;
        rem=rem%5;
        total=quarter+dime+nickel+rem;
        printf("%d\n",total);
    }while(f<0);
    return 0;
}

File: pset6/greedy.py
import cs50
def main():
    while True:
        print("O hai! How much change is owed?")
        f=cs50.get_float()
        if f<0:
            break
        sum=int(round(f*100))
        quarter=sum//25
        rem=sum%25
        dime=rem//10
        rem=rem%10
        nickel=rem//5
        rem=rem%5
        total=quarter+dime+nickel+rem
        print("{}".format(total))
if __name__ == "__main__":
    main()